### PR TITLE
22446 Allow AR corrections (depending on FF)

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#22446

*Description of changes:*
- app version = 7.3.18
- allowed Annual Report corrections for specified legal types only (per FF)
- added new FF with default value

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Dashboard/FilingHistoryList/HeaderActions.vue [hunk 1]
```diff
@@ -142,7 +144,10 @@ export default class HeaderActions extends Mixins(AllowableActionsMixin) {
         if (!this.isBaseCompany) return true
         return false
       case EnumUtilities.isTypeAmalgamationOut(this.filing): return true // not supported
-      case EnumUtilities.isTypeAnnualReport(this.filing): return true // not supported
+      case EnumUtilities.isTypeAnnualReport(this.filing):
+        // enable AR corrections for specified legal types only
+        if (GetFeatureFlag('supported-ar-correction-entities').includes(this.getLegalType)) return false
+        return true
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/HeaderActions.vue:150]: If the feature flag include the current legal type (eg, "CP") then don't disable the "File a Correction" menu item.

File: src/utils/feature-flags.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ const defaultFlagSet: LDFlagSet = {
   'supported-agm-location-chg-entities': [],
   'supported-amalgamation-entities': [],
   'supported-amalgamation-out-entities': [],
+  'supported-ar-correction-entities': [],
```
Comment (severinbeauvais) [inline src/utils/feature-flags.ts:21]: Default value for the FF if LaunchDarkly cannot be reached: no entities are allowed to do AR corrections.

Currently:
- in Dev, the FF is "CP"
- in Test, the FF is "CP"
- in Prod, the FF is ""

File: src/components/Dashboard/FilingHistoryList/HeaderActions.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/Dashboard/FilingHistoryList/HeaderActions.vue:150]: Screenshot (from Dev):

image

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-687-2aewcl50.web.app

**Review by ArwenQin — APPROVED** [review]
Looks good!

**Pull request merged**